### PR TITLE
Add cps-iolib Ver2.0.0 

### PR DESCRIPTION
**Pull request by Syunsuke-Okamoto**

### CODE REVIEW

File: cps-drivers/aio/cpsaio.c [hunk 1]
```diff
@@ -2353,7 +2355,7 @@ static long cpsaio_ioctl( struct file *filp, unsigned int cmd, unsigned long arg
 					spin_lock_irqsave(&dev->lock, flags);
 					valw = (unsigned short) d_ioc.val;
 					DEBUG_CPSAIO_COMMAND(KERN_INFO"DIRECT OUTPUT: [%lx]=%x\n",(unsigned long)( dev->baseAddr + d_ioc.addr ), valw );
-					cps_common_outw( (unsigned long)( dev->baseAddr + d_ioc.addr ) , valw );
+					contec_mcs341_outw( (unsigned long)( dev->baseAddr + d_ioc.addr ) , valw );
```
Comment (ura1231) [inline cps-drivers/aio/cpsaio.c:2359]: (About interrupt prohibition of lines 2355-2359)
Does it also have the purpose of preventing d_ioc.val from being updated by interrupts?
Interrupts are also disabled in contec_m cs341_outw.
For purposes of preventing unintended updates of cps_common_outw, I thought that lines 2355 and 2359 were unnecessary.

File: cps-drivers/aio/cpsaio.c [hunk 2]
```diff
@@ -2367,7 +2369,7 @@ static long cpsaio_ioctl( struct file *filp, unsigned int cmd, unsigned long arg
 					}
 					spin_lock_irqsave(&dev->lock, flags);
 					
-					cps_common_inpw( (unsigned long)( dev->baseAddr + d_ioc.addr ) , &valw );
+					contec_mcs341_inpw( (unsigned long)( dev->baseAddr + d_ioc.addr ) , &valw );
 					DEBUG_CPSAIO_COMMAND(KERN_INFO"DIRECT INPUT:[%lx]=%x\n",(unsigned long)( dev->baseAddr + d_ioc.addr ), valw );
 					d_ioc.val = (unsigned long) valw;
```
Comment (ura1231) [inline cps-drivers/aio/cpsaio.c:2376]: If d_ioc.val has the possibility of being updated by an interrupt, I feel the need to call spin_unlock_irqrestore after copy_to_user () on line 2378.

**Review by ura1231 — COMMENTED** [review]
I listed two points of confirmation.Please confirm.

File: cps-drivers/aio/cpsaio.c (hunk 1 above)
Comment (Syunsuke-Okamoto) [reply cps-drivers/aio/cpsaio.c:2359]: I was understanding.
I change from spin_lock_irqsave function to spin_lock function in IOCTL_CPSAIO_DIRECT_OUTPUT.(and IOCTL_CPSAIO_DIRECT_INPUT).

And , I change from spin_unlock_irqrestore function to spin_unlock function.

File: cps-drivers/aio/cpsaio.c (hunk 2 above)
Comment (Syunsuke-Okamoto) [reply cps-drivers/aio/cpsaio.c:2376]: I was understanding.
I change from spin_lock_irqsave function to spin_lock function in IOCTL_CPSAIO_DIRECT_OUTPUT.(and IOCTL_CPSAIO_DIRECT_INPUT).

And , I change from spin_unlock_irqrestore function to spin_unlock function.

Comment (Syunsuke-Okamoto) [reply cps-drivers/aio/cpsaio.c:2376]: You said, "Do you need to spinlock before copy_from_user (and copy_to_user)?"
I made new issue #12.
I am researching a new branch.

**Review by ura1231 — APPROVED** [review]
There is no problem.
Procees, please.

**Pull request merged**